detekt/wrapper/src/main/java/io/buildfoundation/bazel/detekt/ExecutionUtils.java

```diff
@@ -0,0 +1,117 @@
+package io.buildfoundation.bazel.detekt;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class ExecutionUtils {
+    /**
+     * Returns true if run-as-test-target flag is included in arguments
+     */
+    public static boolean shouldRunAsTestTarget(List<String> arguments) {
+        return arguments.contains("--run-as-test-target");
+    }
+
+    /**
+     * Retrieves the output path for the test result from the input arguments.
+     */
+    public static String getRequiredArgumentValue(List<String> arguments, String argumentName) {
+        String outputPath = getValueForArgumentName(arguments, argumentName);
+        if (outputPath == null) {
+            throw new IllegalStateException("Value not found for argument " + argumentName);
+        }
+        return outputPath;
+    }
+
+    /**
+     * Writes the execution result to a file
+     */
+    public static void writeExecutionResultToFile(Integer exitCode, Path executionResultOutputPath) {
+        try {
+            Files.write(executionResultOutputPath, exitCode.toString().getBytes(), StandardOpenOption.CREATE);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static boolean isParamsFile(String argument) {
+        return argument.startsWith("@");
+    }
+
+    /**
+     * Read-in arguments from a params-file
+     */
+    public static List<String> readArgumentsFromFile(String filePath) {
```

Comment:
```suggestion
    public static List<String> readArgumentsFromFile(Path filePath) {
```

Should also be a Path :)